### ISSUE TITLE
Add `@game(steam: 550)`-like Directive for Launch-Integration

### ISSUE THREAD

**Issue by JoeDailey** [jonah-dev/lfd2-bot#55]
Currently only the SCBL4D plugin supports this feature. We should add a new directive for setting it. It also needs to be integrated with the config print out. 

***Tricky:*** Discord does not support non-http(s) URL links. This makes the game launcher portion of this harder. We have to use a http(s) url to launch the user's web browser and that website needs to redirect to [`steam://run/<id>//<args>/`](https://developer.valvesoftware.com/wiki/Steam_browser_protocol#:~:text=steam://run/%3Cid%3E//%3Cargs%3E/)

This feature currently just powers the "Click here to launch the game" link when the lobby is full. But in the future we can define a big list of presets for every game. Here's an example... 

Dead by Daylight is a 1v4 pvp game. It's steam ID is 381210. The directive `@game(steam: 381210)` would lookup 381210 in our game dictionary and set the lobby settings accordingly, e.g. vTeams=[1,4].

**Comment by celowsky** [comment]
It sounds like this solution could be implemented using Route53 (AWS), Google Domains, or another DNS registrar with a decent API to generate subdomains that can redirect with the proper steam link. We just need to come up with an interface for how to create and manage these redirects (or use the default provided tools)

**Comment by JoeDailey** [comment]
Nice, sounds like you have ideas. My naïve approach would be to create an interface like you suggested, e.g. `http://<ip>/steam/run/<id>//<args>/`. The response is a [300 status code with the `Location` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Redirections) set to the equivalent steam protocol url. 

Perhaps there's are more clever way. Perhaps this has issues unseen. Feel free to do whatever works. Let's aim for low maintenance.

**Comment by JoeDailey** [comment]
Alternatively, I'm surprised this doesn't exist. If there's already a public API that does this we could save ourselves a lot of complexity and maintenance costs.

**Comment by celowsky** [comment]
I think what's really missing here is a game management interface. I agree that creating subdomains that redirect to the proper steam url is the way to go. But how do we add a new game redirect? For LFD2, I did so by hardcoding, just to see if it'd work. But going forward, I'd like to automate the process as much as possible.

**Comment by celowsky** [comment]
Maybe its enough to start out with an admin panel (through Google, ,AWS, or whoever) that controls the redirects. But it'd be ideal if could let anyone add they're game if its not already in the bot. Provided they have the proper access control. Something else I need to think about.

**Comment by JoeDailey** [comment]
A good-enough incremental step would just be a hard-coded dictionary of steam-ID => settings list. I really don't think we should invest in anything more until we see how much this is used. If we wanted to go with a more DB-like approach I think a YAML flatfile would be better than a local or remote DB. A text file is technically and interface, right? Perhaps someday something more impressive will be worth the complexity.

**Comment by JoeDailey** [comment]
To recap there's two workstreams.

- Creating a `@games` directive to take steam ID. Cross-list it with a repository of game settings. Update the lobby settings accordingly. 
- Solving the steam protocol issue so that we can add the 'launch' link to more games than L4D2

**Comment by JoeDailey** [comment]
@celowsky, I did some testing. It turns out that steam browser protocols are supported. The only issue is that you cannot use them with markdown. E.g. `[launch game](steam://run/550)` will be rendered without the hyperlink as "launch game". I feel like this is an oversight on Discords end. I also tried setting the `url` field on the Embed object to this URL but that is also blocked by a 550 response from Discord's API. I think it's because their web crawler doesn't support non-http URLs. (understandable)

So I think the simplest solution worth using is just rendering the steam URL raw in the embed message. It's ugly but it's cheap. In the meantime we can open a request for discord to fix the markdown parser.
